blog/_posts/2024-08-19-KAITO-finetuning.md

```diff
@@ -0,0 +1,139 @@
+
+> If you are curious to know how the total steps are calculated, here is the formula: 
+> `total steps = number of epochs * (number of samples in dataset / batch size)`, where the number of epochs and batch size are configurable parameters. 
+
+KAITO allows users to apply a custom ConfigMap to overwrite most of the tuning parameters used by the tuning job. We should be careful about changing those parameters though. For example, increasing the batch size, i.e., leveraging higher data parallelism, will reduce the tuning time but use more GPU memory, and require better GPUs in general. 
+
+When the fine-tuning job is complete, the result, which is often referred to as an **adapter**, will be packaged as a container image and stored in the specified output location. The adapter image is lightweight, portable, conveniently version controlled, and can be pulled into a new inferencing service!
+
+After submitting your tuning workspace manifest, go get yourself a drink or a snack. 😉
+
+![Three hours later](/blog/assets/images/three_hours_later_image.png)
+
+## Now that tuning is done, how can we quickly see the result?
+
+To use our new fine-tuned adapter, the inference workspace will now accept one or more adapters to customize model behavior.
+
+Here is an example of what a fine-tuned inference workspace looks like:
+
+```bash
+apiVersion: kaito.sh/v1alpha1
+kind: Workspace
+metadata:
+  name: workspace-phi-3-mini-adapter
+resource:
+  instanceType: Standard_NC6s_v3
+  labelSelector:
+    matchLabels:
+      apps: phi-3-adapter
+inference:
+  preset:
+    name: phi-3-mini-128k-instruct
+  adapters:
+    - source:
+        name: myadapter
+        image: myregistry.azurecr.io/adapters/myadapter:0.0.1
+        imagePullSecrets:
+          - myregistrysecret
+      strength: "1.0"
+```
+
+As new inference requests come in, they’ll flow through the model that merges the adapters and the response will be affected by the newly trained data. 
+Great! After updating our inference endpoint to leverage the new adapter, let’s ask the same question and check the response from the fine-tuned phi-3-mini-128k-instruct model:
+
+
+![Screenshot of tuned model response](/blog/assets/images/kaito_tuned_phi3_response.png)
+*Screenshot of tuned model response*
+
+That definitely looks more accurate – the fine-tuned model quickly discerned that we’re looking for an answer related to Kubernetes and provided a much better response. However, there’s always room to improve the model’s accuracy for domain specific questions. We can improve upon or find new input datasets to create new tuning adapters. This iterative tuning process can be done by repeating the steps described above.
+
+## Other options to make models smarter
```

Comment:
To my question above, this might warrant an in-page link from above. And no strict need to put this in the end if it makes more sense when you introduce the method